libnd4j/include/array/cpu/NDArray.cpp

```diff
@@ -97,10 +93,10 @@ void NDArray::fillAsTriangular(const float val, int lower, int upper, NDArray& t
     for (auto i = start; i < stop; i++) {
       shape::index2coordsCPU(start, i, target.shapeInfo(), coords);
       const auto zOffset = shape::getOffset(target.shapeInfo(), coords);
-
       // if( (row + upper < col) || (row + lower > col) )
-      if ((coords[zRank - 2] + upper < coords[zRank - 1]) || (coords[zRank - 2] + lower > coords[zRank - 1]))
+      if (coords[0] > (coords[1] - lower)) {
```

Comment:
explicit indexing  will work only for rank==2. is it preconditioned? otherwise I would go with the previous implementation